packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -499,6 +501,24 @@ Future<void> buildGradleApp({
     '$successMark Built ${globals.fs.path.relative(apkFile.path)}$appSize.',
     color: TerminalColor.green,
   );
+
+  // Call size analyzer if --analyze-size flag was provided.
+  if (buildInfo.analyzeSize != null && !globals.platform.isWindows) {
+    final SizeAnalyzer sizeAnalyzer = SizeAnalyzer(
+      fileSystem: globals.fs,
+      logger: globals.logger,
+      processUtils: ProcessUtils.instance,
+    );
+    final Map<String, Object> output = await sizeAnalyzer.analyzeApkSizeAndAotSnapshot(
+      apk: apkFile,
+      aotSnapshot: globals.fs.file(buildInfo.analyzeSize),
+    );
+    final File outputFile = globals.fs.file('apk-analysis.json')
```

Comment:
Could we also make this a unique file?

Reply:
Sure, how do you want me to surface that info to devtools? Or does it rely on the user dropping it into the UI?

Reply:
It would rely on the user dropping it into the UI. I just meant using `globals.fsUtils.getUniqueFile();` to create the file if that clarifies!

Reply:
Done